File: src/main/java/com/revature/egghead/model/Egg.java
package com.revature.egghead.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;

import javax.persistence.*;
import java.sql.Timestamp;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Egg {
    @Id
    @GeneratedValue
    private int id;

    private boolean isReady;
    @CreationTimestamp
    private Timestamp timeCreated;
    private Timestamp timeComplete;
    private int startingSize;
    private int maxSize;
    private int currentSize;
    private int redValue;
    private int greenValue;
    private int blueValue;
    private String animalType;

    @ManyToOne
    @JoinColumn(name = "incubator_id")
    private Incubator incubator;
}

File: src/app/models/egg.ts
export interface Egg {
    id?:number;
    ready:boolean;
    timeCreated:string;
    timeComplete:string;
    startingSize?:number;
    maxSize?:number;
    currentSize:number;
    redValue:number;
    greenValue:number;
    blueValue:number;
    animalType:string;
}